cli/src/cmd/build.rs

```diff
@@ -88,14 +92,31 @@ pub fn collect_crate_metadata() -> Result<CrateMetadata> {
     })
 }
 
-/// Invokes `cargo build` in the current directory.
+/// Invokes `cargo build` in the specified directory, defaults to the current directory.
 ///
 /// Currently it assumes that user wants to use `+nightly`.
-fn build_cargo_project() -> Result<()> {
-    // We also assume that the user uses +nightly.
-    let output = Command::new("cargo")
+fn build_cargo_project(working_dir: Option<&PathBuf>) -> Result<()> {
+    let mut cmd = Command::new("cargo");
+    let mut is_nightly_cmd = Command::new("cargo");
+    if let Some(dir) = working_dir {
+        cmd.current_dir(dir);
+        is_nightly_cmd.current_dir(dir);
+    }
+
+    let is_nightly_default = is_nightly_cmd
+        .arg("--version")
+        .output()
+        .map_err(|_| ())
+        .and_then(|o| String::from_utf8(o.stdout).map_err(|_| ()))
+        .unwrap_or_default()
+        .contains("-nightly");
+
+    if !is_nightly_default {
+        cmd.arg("+nightly");
+    }
+
+    let output = cmd
         .args(&[
-            "+nightly",
```

Comment:
I am still undecided about this because for CI it makes total sense but not for all people that are generally working with stable Rust.